sycl/test/check_device_code/matrix/matrix-nvptx-tf32-test.cpp

```diff
@@ -0,0 +1,141 @@
+          joint_matrix_load(sg, sub_c, accC.get_pointer(), N);
+
+          // CHECK: tail call i32 @llvm.nvvm.f2tf32.rna(float {{.*}}
+          // Round a, b to tf32
+          for (auto i = 0; i < 4; ++i)
+            sub_a.data[i] = float_to_tf32(sub_a.data[i]);
+
+          for (auto i = 0; i < 4; ++i)
+            sub_b.data[i] = float_to_tf32(sub_b.data[i]);
+
+          //CHECK: tail call { float, float, float, float, float, float, float, float } @llvm.nvvm.wmma.m16n16k8.mma.row.row.tf32(i32 {{.*}}, i32 {{.*}}, i32 {{.*}}, i32 {{.*}}, i32 {{.*}}, i32 {{.*}}, i32 %{{.*}}, i32 {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}) #{{.*}}
+          sub_c = joint_matrix_mad(sg, sub_a, sub_b, sub_c);
+          //CHECK: tail call void @llvm.nvvm.wmma.m16n16k16.store.d.row.stride.f32.p1f32(float addrspace(1)* {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}, float {{.*}}, i32 {{.*}}
+          joint_matrix_store(sg, sub_c, accD.get_pointer(), N);
+        });
+  });
+
+  q.submit([&](handler &cgh) {
+    auto accA = bufA.get_access<access::mode::read_write>(cgh);
+    auto accB = bufB.get_access<access::mode::read_write>(cgh);
+    auto accC = bufC.get_access<access::mode::read_write>(cgh);
+    auto accD = bufD.get_access<access::mode::read_write>(cgh);
+
+    cgh.parallel_for<class col_col>(
+        nd_range<2>({1, 32}, {1, 32}),
+        [=](nd_item<2> item) [[sycl::reqd_work_group_size(1, 1, 32)]] {
+          sycl::sub_group sg = item.get_sub_group();
+
+          joint_matrix<precision::tf32, matrix_use::a, M, K,
+                       matrix_layout::col_major>
+              sub_a;
+
+          joint_matrix<precision::tf32, matrix_use::b, K, N,
+                       matrix_layout::col_major>
+              sub_b;
+
+          joint_matrix<float, matrix_use::accumulator, M, N,
+                       matrix_layout::col_major>
+              sub_c;
+
+          //CHECK: tail call { i32, i32, i32, i32 } @llvm.nvvm.wmma.m16n16k8.load.a.col.stride.tf32.p0i32(i32* %call.ascast.i.i{{.*}}.i, i32 8) #{{.*}}
+          joint_matrix_load(sg, sub_a, accA.get_pointer(), K);
+          //CHECK: tail call { i32, i32, i32, i32 } @llvm.nvvm.wmma.m16n16k8.load.b.col.stride.tf32.p0i32(i32* %call.ascast.i.i{{.*}}.i, i32 16) #{{.*}}
+          joint_matrix_load(sg, sub_b, accB.get_pointer(), N);
+          //CHECK: tail call { float, float, float, float, float, float, float, float } @llvm.nvvm.wmma.m16n16k16.load.c.col.stride.f32.p1f32(float addrspace(1)* {{.*}}, i32 {{.*}}) #{{.*}}
+          joint_matrix_load(sg, sub_c, accC.get_pointer(), N);
+
+          // CHECK: tail call i32 @llvm.nvvm.f2tf32.rna(float {{.*}}
+          // Round a, b to tf32
+          for (auto i = 0; i < 4; ++i)
+            sub_a.data[i] = float_to_tf32(sub_a.data[i]);
```

Comment:
You seem to miss updating the name to round_to_tf32  in the test case.

Reply:
Sorry about that. Updated now!